### ISSUE TITLE
Minor ticks on log-scale colorbar are not cleared

### ISSUE THREAD

**Issue by lamorton** [matplotlib/matplotlib#8358]
### Bug report

**Bug summary**

- Using pcolormesh with logarithmic color scale and minor ticks enabled results in a cluttered-looking set of minor ticks, apparently because the linear ticks are not removed, just overplotted with the log scale ones.  

**Code for reproduction**

```python
import numpy
from matplotlib.colors import LogNorm
from matplotlib.pyplot import pcolormesh, colorbar, rcParams
rcParams['xtick.minor.visible']=True
rcParams['ytick.minor.visible']=True
data=numpy.random.random((10,12))*100.0+0.1 
pcolormesh(data,norm=LogNorm())
colorbar()
```

**Actual outcome**
![lognormmess](https://cloud.githubusercontent.com/assets/23484003/24163864/247ba266-0e42-11e7-8a75-6ff4fe34bbb3.png)

**Expected outcome**

- Expect that the colobar minor ticks would appear logarithmically.

**Version Info**
Platform: GNU/Linux 2.6.32-642.11.1.el6.x86_64  
Python: 2.7.13 |Anaconda custom (64-bit)
IPython: 5.2.2
Matplotib: 2.0.0

All python components installed via Anaconda.

**Comment by phobson** [comment]
Thanks for the detailed report. I can reproduce this.

For the time being, here's how you can somewhat work around this:

```python
import numpy
from matplotlib.colors import LogNorm
from matplotlib.pyplot import pcolormesh,colorbar, rcParams
rcParams['xtick.minor.visible']=True
rcParams['ytick.minor.visible']=True
data = numpy.random.random((10,12))*100.0+0.1 
img = pcolormesh(data,norm=LogNorm())
cbar = colorbar(img)
cbar.ax.minorticks_off()
```

**Comment by billtau** [comment]
I just upgraded to matplotlib 2.0.1, and I still see the behavior described by PO.

**Comment by afvincent** [comment]
FWIW, this issue is still present on Matplotlib 2.1 (and master `2.1.0.post357+g075618bb4`). (@phobson's workaround is still fine though)

**Comment by Dih5** [comment]
This has been fixed by #9903.
Here is a figure obtained right now using the master branch:
![figure_1](https://user-images.githubusercontent.com/12070738/37824103-d9fedf9e-2e8b-11e8-8b16-a91c016f1465.png)

**Comment by jklymak** [comment]
Great!
